sci_strategic_plugin/src/sci_strategic_plugin.cpp

```diff
@@ -257,59 +330,153 @@ bool SCIStrategicPlugin::planManeuverCb(cav_srvs::PlanManeuversRequest& req, cav
+
+      if (case_num < 3)
+      {
+        resp.new_plan.maneuvers.push_back(composeLaneFollowingManeuverMessage(
+          case_num, current_downtrack_, stop_intersection_down_track-config_.stop_line_buffer, current_state.speed, 0.0,
+          current_state.stamp, time_to_schedule_stop,
+          lanelet::utils::transform(crossed_lanelets, [](const auto& ll) { return ll.id(); })));
+
+      }
+      else
+      {
+        ROS_DEBUG_STREAM("Decelerating to stop at the intersection");
+
+        // at the stop line or decelerating to stop line
+        // stop and wait maneuver
+       
+        std::vector<lanelet::ConstLanelet> crossed_lanelets =
+            getLaneletsBetweenWithException(current_downtrack_, stop_intersection_down_track, true, true);
+        
+
+        double stopping_accel = caseThreeSpeedProfile(stop_intersection_down_track-current_state.downtrack, current_state.speed, time_to_schedule_stop);
+        // TODO: temp val for acc- remove when carma streets is running
+        stopping_accel = 1.5;
+        resp.new_plan.maneuvers.push_back(composeStopAndWaitManeuverMessage(
+          current_state.downtrack, stop_intersection_down_track-config_.stop_line_buffer, current_state.speed, crossed_lanelets[0].id(),
+          crossed_lanelets[0].id(), stopping_accel, current_state.stamp,
+          current_state.stamp + ros::Duration(time_to_schedule_stop)));
+
+        
+      }
+  }
 
-      resp.new_plan.maneuvers.push_back(composeLaneFollowingManeuverMessage(
-        case_num, current_downtrack_, stop_intersection_down_track, current_state.speed, 0.0,
-        current_state.stamp, time_to_schedule_stop,
-        lanelet::utils::transform(crossed_lanelets, [](const auto& ll) { return ll.id(); })));
-    }
-    else
-    {
-      // at the stop line
-      // stop and wait maneuver
-      auto stop_line_lanelet = nearest_stop_intersection->lanelets().front();
-      double stop_duration = (scheduled_enter_time_ - scheduled_stop_time_)*1000.0;
-      ROS_DEBUG_STREAM("Planning stop and wait maneuver");
-      resp.new_plan.maneuvers.push_back(composeStopAndWaitManeuverMessage(
-        current_downtrack_, stop_intersection_down_track, current_state.speed, stop_line_lanelet.id(),
-        stop_line_lanelet.id(), current_state.stamp,
-        current_state.stamp + ros::Duration(stop_duration)));
+  bool cond1 = (int(street_msg_timestamp_) - int(scheduled_stop_time_)) > 0;
+  bool cond2 = (int(street_msg_timestamp_) - int(scheduled_enter_time_)) < 0;
+  bool time_to_stop_at_stopline = (cond1 && cond2);
+  ROS_DEBUG_STREAM("time_to_stop_at_stopline: " << time_to_stop_at_stopline);
 
-    }
+  if (time_to_stop_at_stopline)
+  {
+    base_time = std::max(scheduled_stop_time_, street_msg_timestamp_);
+    double stop_duration = (scheduled_enter_time_ - base_time)/1000;
+    ROS_DEBUG_STREAM("stop_duration:  " << stop_duration);
+    ROS_DEBUG_STREAM("Planning stop and wait maneuver");
+    double stopping_accel = 1.5;
```

Comment:
Should this be parameterized? Either for this plugin or based on the vehicle's decelearation limits?

Reply:
At this point, the vehicle is already stopped so we just need a valid acceleration value. The value doesnt affect the performance. But made it configurable.